receiver/skywalkingreceiver/internal/metrics/metric_report_service.go

```diff
@@ -52,18 +53,21 @@ func NewReceiver(nextConsumer consumer.Metrics, set receiver.CreateSettings) (*R
 
 // Collect implements the service Collect traces func.
 func (r *Receiver) Collect(ctx context.Context, jvmMetricCollection *agent.JVMMetricCollection) (*common.Commands, error) {
-	err := consumeMetrics(ctx, jvmMetricCollection, r.nextConsumer)
+	err := consumeMetrics(ctx, jvmMetricCollection, r.nextConsumer, r.grpcObsrecv)
 	if err != nil {
 		return &common.Commands{}, err
 	}
 	return &common.Commands{}, nil
 }
 
-func consumeMetrics(ctx context.Context, collection *agent.JVMMetricCollection, nextConsumer consumer.Metrics) error {
+func consumeMetrics(ctx context.Context, collection *agent.JVMMetricCollection, nextConsumer consumer.Metrics, obsreport *receiverhelper.ObsReport) error {
 	if collection == nil {
 		return nil
 	}
 	pmd := SwMetricsToMetrics(collection)
-	return nextConsumer.ConsumeMetrics(ctx, pmd)
+	obsContext := obsreport.StartMetricsOp(ctx)
```

Comment:
I am not sure, but should the metrics ops before above line 67?

Reply:
StartMetricsOp will create span to track the operation.  so it  makes sense to track consumerMetric?